### PR TITLE
Support ngx_mgmt_module

### PR DESCRIPTION
**Pull request by dakshinai**
### Proposed changes

Adding support for `ngx_mgmt_module`

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [`CONTRIBUTING`](https://github.com/nginxinc/nginx-go-crossplane/blob/main/CONTRIBUTING.md) document
- [x] If applicable, I have added tests that prove my fix is effective or that my feature works
- [x] If applicable, I have checked that any relevant tests pass after adding my changes
- [ ] I have updated any relevant documentation ([`README.md`](https://github.com/nginxinc/nginx-go-crossplane/blob/main/README.md))

### CODE REVIEW

File: analyze.go [hunk 1]
```diff
@@ -52,10 +52,11 @@ const (
 	ngxHTTPSifConf    = 0x20000000 // http > server > if
 	ngxHTTPLifConf    = 0x40000000 // http > location > if
 	ngxHTTPLmtConf    = 0x80000000 // http > location > limit_except
+	ngxMgmtMainConf   = 0x00000002 // mgmt // TODO: NLB-4310 set correct value here
```
Comment (dakshinai) [inline analyze.go:55]: TBD: to figure out correct value

Comment (dakshinai) [reply analyze.go]: Set a unique mask w.r.t crossplane

File: analyze.go [hunk 2]
```diff
@@ -1836,6 +1868,9 @@ var directives = map[string][]uint{
 	"upstream_conf": {
 		ngxHTTPLocConf | ngxConfNoArgs,
 	},
+	"usage_report": {
+		ngxMgmtMainConf | ngxConfAny,
```
Comment (dakshinai) [inline analyze.go:1872]: NGINX does accept `usage_report;`

Comment (ornj) [reply analyze.go:1872]: Then should it be `ngxConfNoArgs | ngxConfTake12`?

File: analyze.go [hunk 3]
```diff
@@ -29,6 +29,7 @@ const (
 	ngxConf2More = 0x00002000 // >=2 args
 
 	// some helpful argument style aliases.
+	ngxConfTake012  = ngxConfNoArgs | ngxConfTake1 | ngxConfTake2
```
Comment (ornj) [inline analyze.go:32]: These mostly match the variables listed in the [Development Guide](https://nginx.org/en/docs/dev/development_guide.html#config_directives) (not values, but variables). We don't export these yet but I think it makes sense to do so soon. I'm not so sure we need to add another "alias" and should just do `ngxConfNoArgs | ngxConfTake1 | ngxConfTake2`  in the directive mapping.

**Review by ornj — APPROVED** [review]

**Review by xynicole — APPROVED** [review]

**Pull request merged**